### PR TITLE
[CARBONDATA-2277] fix for filter of default values on all datatypes

### PR DESCRIPTION
**Pull request by jatin9896**
1. Added solution to handle filter keys for the direct dictionary on default values.
2. For no dictionary columns changes code to get correct bytes value of default values.
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? No 
 
 - [ ] Any backward compatibility impacted? No
 
 - [ ] Document update required? No

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required? Yes
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4575/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3349/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4073/

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/AlterTableValidationTestCase.scala [hunk 1]
```diff
@@ -584,6 +589,98 @@ test("test alter command for boolean data type with correct default measure valu
...
+        sql(
+          "alter table test add columns (c3 timestamp) TBLPROPERTIES('DEFAULT.VALUE.c3' = " +
+          "'1996/01/01 11:11:11', 'DICTIONARY_INCLUDE' = 'c3')")
+      } else {
+        sql(
+          "alter table test add columns (c3 timestamp) TBLPROPERTIES('DEFAULT.VALUE.c3' = " +
+          "'1996/01/01 11:11:11')")
+      }
+      println("Timestamp Format: " + CarbonProperties.getInstance().getProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT))
+      println("flag : "+ flag)
```
Comment (manishgupta88) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/AlterTableValidationTestCase.scala]: Remove all println statements

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/AlterTableValidationTestCase.scala [hunk 2]
```diff
@@ -584,6 +589,98 @@ test("test alter command for boolean data type with correct default measure valu
...
+        "insert into test select 4,'String1',12345,'area',20,'country','2017-02-12','1996/01/01 " +
+        "10:10:10'")
+      checkAnswer(sql("select id from test where c3='1996-01-01 11:11:11'"), Seq(Row(1)))
+      checkAnswer(sql("select id from test where c3<'1995-01-01 11:11:11'"), Seq(Row(2)))
+      checkAnswer(sql("select id from test where c3>'1994-01-02 11:11:11'"),
+        Seq(Row(3), Row(4), Row(1)))
+      checkAnswer(sql("select id from test where c3>='1995-01-01 11:11:11'"),
+        Seq(Row(3), Row(4), Row(1)))
+      checkAnswer(sql("select id from test where c3<='1995-01-02 11:11:11'"), Seq(Row(2),Row(3)))
+      CarbonProperties.getInstance()
```
Comment (manishgupta88) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/AlterTableValidationTestCase.scala]: enclose the test case in try and finally to ensure that the property getting set in try block gets unset in finally block

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/AlterTableValidationTestCase.scala [hunk 3]
```diff
@@ -584,6 +589,98 @@ test("test alter command for boolean data type with correct default measure valu
...
+        sql(s"alter table test add columns (c3 int) TBLPROPERTIES('DEFAULT.VALUE.c3' = '23')")
+      }
+      sql("insert into test select 2,'String1',12345,'area',5000,'country','2017/02/12',25")
+      sql("insert into test select 3,'String1',12345,'area',5000,'country','2017/02/12',35")
+      sql("insert into test select 4,'String1',12345,'area',5000,'country','2017/02/12',45")
+      checkAnswer(sql("select id from test where c3=23"), Seq(Row(1)))
+      checkAnswer(sql("select id from test where c3<34"), Seq(Row(1), Row(2)))
+      checkAnswer(sql("select id from test where c3>24"), Seq(Row(2), Row(3), Row(4)))
+      checkAnswer(sql("select id from test where c3>=35"), Seq(Row(3), Row(4)))
+      checkAnswer(sql("select id from test where c3<=35"), Seq(Row(1), Row(2), Row(3)))
```
Comment (manishgupta88) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/restructure/AlterTableValidationTestCase.scala]: Add a query for !=, null and not null cases also

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3416/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4151/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4641/

**Comment by manishgupta88** [comment]
LGTM

**Pull request closed without merging**